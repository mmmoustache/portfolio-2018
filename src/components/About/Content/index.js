import styled from "styled-components";
import { Breakpoints } from '../../../utils/breakpoints';
import { Timings, Durations } from '../../../utils/transitions';
import { convertRemToPixels as rem } from '../../../utils/utils';

const Medium = {
  paddingLeft: `${rem('60px')}`,
  widthOffset: `${rem('360px')}`,
};

const offset = `${rem('20px')}`;

export default styled.div`
  opacity: 0;
  transform: translateY(${offset});
  transition-delay: ${Durations.slow};
  transition: opacity ${Durations.default} ${Timings.default}, transform ${Durations.default} ${Timings.default};
  
  @media ${Breakpoints.medium} {
    padding-left: ${Medium.paddingLeft};
    width: calc(100% - ${Medium.widthOffset});
  }

  ${
    props => props.isVisible && `
      opacity: 1;
      transform: translateY(0);
    `
  }
`;
