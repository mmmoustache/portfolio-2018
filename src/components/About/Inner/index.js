import styled from "styled-components";
import { Breakpoints } from '../../../utils/breakpoints';
import { Layouts } from '../../../utils/layouts';
import { convertRemToPixels as rem } from '../../../utils/utils';

export default styled.div`
  box-sizing: border-box;
  margin: 0 auto;
  max-width: ${Layouts.constraint};
  padding: ${rem('80px')} ${rem('20px')};
  
  @media ${Breakpoints.medium} {
    align-items: center;
    display: flex;
    justify-content: center;
  }
`;
