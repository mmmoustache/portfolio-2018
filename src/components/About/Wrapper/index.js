import styled from "styled-components";
import { Colours } from '../../../utils/colours';

export default styled.div`
  background: ${Colours.white};
  width: 100%;
`;
