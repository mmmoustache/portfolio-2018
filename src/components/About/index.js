import React from 'react';
import H2 from '../H2';
import H3 from '../H3';
import Text from "../Text";
import Avatar from "../Avatar";
import Content from "./Content";
import Wrapper from "./Wrapper";
import Inner from "./Inner";

const About = ({ data, isVisible }) => 
  <Wrapper id="about">
    <Inner>
      <Avatar isVisible={isVisible} />
      <Content isVisible={isVisible}>
        <H3>{data.subtitle}</H3>
        <H2>{data.title}</H2>
        <Text>{ data.text && data.text.map(e => <p>{e}</p>) }</Text>
      </Content>
    </Inner>
  </Wrapper>
;

export default About;
