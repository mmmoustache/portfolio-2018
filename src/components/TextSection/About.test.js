import React from 'react';
import renderer from 'react-test-renderer';
import About from '../About';

describe('Component: About', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <About data={{}} isVisible />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
