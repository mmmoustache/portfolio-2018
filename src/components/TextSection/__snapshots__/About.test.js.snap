// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: About Should render successfully 1`] = `
<div
  className="sc-EHOje hGGpUi"
  id="about"
>
  <div
    className="sc-bZQynM fdYyUr"
  >
    <div
      className="sc-bxivhb bsRDV"
    />
    <div
      className="sc-ifAKCX fotatZ"
    >
      <h3
        className="sc-bwzfXH ejbyUv"
      />
      <h2
        className="sc-bdVaJa kujZxV"
      />
      <div
        className="sc-htpNat fjAnuH"
      />
    </div>
  </div>
</div>
`;
